### ISSUE TITLE
Security of the solution?

### ISSUE THREAD

**Issue by TejasAntani** [dgreif/ring#216]
Hi - this is a more fundamental question for my own knowledge. 

Are there any known security risks that I introduce by enabling this plugin versus the standard Rong app? Is there any increased likelihood that a bad actor could gain access to my ring system?

Is there a more secure way to configure Homebridge and the ring plugin to reduce these risks? I’m using the 2FA method and homebridge is running on a Mac. 

Thanks for the education!

**Comment by htims1989** [comment]
TL;DR keep the refresh token safe by limiting who can access it and you'll be fine.

So long as you keep the refresh token safe, there shouldn't be a problem. It's worth mentioning that when you use many websites your web browser holds onto a refresh token (or a cookie which is pretty much the same thing in this context) in order to keep you signed in as you move around the site or away from it, if someone were to get a hold of that they would be able to sign in to your account on those websites. The ring.com website looks to use a session cookie, if someone were to get a hold of that out of your browser they would be able to access the site as you. This is not usually a problem as in order for someone to access your browser, you would have to unlock your machine/device using your password/biometrics. Of course this a massive over simplification ignoring exploits etc but I really wouldn't worry about it too much!

**Comment by dgreif** [comment]
I think @htims1989 is pretty spot-on.  Your homebridge config file should be kept as safe as you possibly can because it contains your auth info for Ring (and presumably other IoT devices as well).  Beyond that, I think running homebridge in "insecure" mode, which is used for things like Alexa integration, is probably the most insecure thing that you can do.  HomeKit seems to take security very seriously, so I don't think there will be many/any vulnerabilities from the Apple side.
